### ISSUE TITLE
macOS 14.3 on vmapple doesn't boot

### ISSUE THREAD

**Issue by frefire** [blacktop/darwin-xnu-build#16]
The KernelCollection compiled successfully. And I disabled SIP and used 'kmutil configure-boot' to setup the new collection. The machine is in UTM with Apple's virtualization.framework. But no matter what I do, when I reboot, it always show a window that says "Custom kernel failed to boot" with no panic log at all. I don't know what I've missed. I've tried every way to use 'kmutil configure-boot' that I could google but nothing works. The result is always the same.

Can you help? Thanks.

**Comment by blacktop** [comment]
Did you create the kernel collection via kmutil in a Virtual.fw VM?

**Comment by frefire** [comment]
Yes, I create the collection on the same machine that I boot-configure on, which is the virtualization.framework based UTM, and I also double checked MACHINE_CONFIG=VMAPPLE

**Comment by jprx** [comment]
Virtualization.framework guests on 14.3 and older will not boot custom kernels without [some modifications](https://gist.github.com/steven-michaud/fda019a4ae2df3a9295409053a53a65c).

Guests running 14.4 and on should allow you to boot custom kernels without these changes.
